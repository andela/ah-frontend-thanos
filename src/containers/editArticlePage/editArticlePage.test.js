import React from 'react';
import { mount, shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import EditArticlePage, { mapDispatchToProps } from './index';

describe('<CreateArticlePage />', () => {
  let EditArticlePageComponent;
  let wrapper;
  beforeEach(() => {
    jest.resetModules();
    const middlewares = [thunk];
    const mockStore = configureStore(middlewares);
    const props = { articleId: 1 };
    const initialState = { editArticleReducer: { articleEditData: { tag_list: 'one,two', title: 1 } }, articleReducer: { article: 'good work' } };
    const store = mockStore(initialState);
    const historyMock = { push: jest.fn() };
    EditArticlePageComponent = mount(
      <EditArticlePage
        store={store}
        articleId={props.articleId}
        history={historyMock}
        match={{ params: { id: 1 } }}
      />,
    );
    wrapper = shallow(
      <EditArticlePage store={store} articleId={props.articleId} />,
    );
  });
  test('test cloudinary function ', async () => {
    global.cloudinary = {
      openUploadWidget: (params, cb) => {
        cb(null, {
          event: 'success',
          info: { secure_url: 'http://cloudinary/img/123.png' },
        });
      },
    };
    const widgetFn = wrapper.dive().instance().OnClickHandler();
    expect(widgetFn);
  });

  it('should render the component', () => {
    expect(EditArticlePageComponent).toMatchSnapshot();
  });
  const dispatchTest = dispatch => (
    mapDispatchToProps(dispatch).editArticle({})
    && mapDispatchToProps(dispatch).editData({})
    && mapDispatchToProps(dispatch).updateImage({})
    && mapDispatchToProps(dispatch).loadEditData({})
  );
  it('should dispatch a method to get user input', () => {
    const dispatch = jest.fn();
    dispatchTest(dispatch);
  });

  it('should test user data without fail', () => {
    const targetData = (nameData, valueData) => (
      { target: { name: nameData, value: valueData } }
    );
    EditArticlePageComponent
      .find('textarea#titleInput')
      .simulate('change', targetData('title', 'How ro go to school'));
    EditArticlePageComponent
      .find('form')
      .simulate('submit');
    EditArticlePageComponent
      .find('input.image')
      .simulate('change', targetData('image_url', 'http://image.com'));
  });
});
