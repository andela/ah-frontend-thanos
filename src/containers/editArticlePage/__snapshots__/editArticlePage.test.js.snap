// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateArticlePage /> should render the component 1`] = `
<Connect(EditArticlePage)
  articleId={1}
  history={
    Object {
      "push": [MockFunction],
    }
  }
  match={
    Object {
      "params": Object {
        "id": 1,
      },
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <EditArticlePage
    article="good work"
    articleId={1}
    editArticle={[Function]}
    editArticleReducer={
      Object {
        "articleEditData": Object {
          "tag_list": "one,two",
          "title": 1,
        },
      }
    }
    editData={[Function]}
    history={
      Object {
        "push": [MockFunction],
      }
    }
    loadEditData={[Function]}
    match={
      Object {
        "params": Object {
          "id": 1,
        },
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    updateImage={[Function]}
  >
    <div>
      <EditArticle
        article="good work"
        onChange={[Function]}
        onClick={[Function]}
        onSubmit={[Function]}
      >
        <div>
          <br />
          <h3
            className="text-center createArticleTitle"
          >
            Edit Your Article
          </h3>
          <hr
            className="col-sm-8 "
          />
          <br />
          <div
            className="col-sm-10 offset-2 artcileCreate"
          >
            <form
              className="create-article-form"
              onSubmit={[Function]}
            >
              <h5>
                Title: 
              </h5>
              <small>
                Title must be 130 characters maximum and 5 characters minimum
              </small>
              <div
                className="form-group"
              >
                <textarea
                  className="form-control"
                  id="titleInput"
                  maxLength="130"
                  minLength="5"
                  name="title"
                  onChange={[Function]}
                  required={true}
                  rows="3"
                />
              </div>
              <h5>
                Description
              </h5>
              <small>
                Title must be 150 characters maximum and 100 characters minimum
              </small>
              <div
                className="form-group"
              >
                <textarea
                  className="form-control"
                  id="descriptionInput"
                  maxLength="120"
                  minLength="100"
                  name="description"
                  onChange={[Function]}
                  required={true}
                  rows="3"
                />
              </div>
              <h5>
                Tag list
              </h5>
              <div
                className="form-group"
              >
                <textarea
                  className="form-control"
                  id="tag_listInput"
                  name="tag_list"
                  onChange={[Function]}
                  required={true}
                  rows="3"
                />
              </div>
              <br />
              <h5>
                Upload Image
              </h5>
              <div
                className="col-lg-8 mt-2"
              >
                <input
                  className="image"
                  name="image_url"
                  onClick={[Function]}
                  type="file"
                />
              </div>
              <br />
              <h5>
                Body
              </h5>
              <div
                className="form-group"
              >
                <textarea
                  className="form-control"
                  id="bodyInput"
                  name="body"
                  onChange={[Function]}
                  required={true}
                  rows="3"
                />
              </div>
              <div
                className="signin-createArticleBt"
              >
                <button
                  className="btn btn-primary"
                  id="createArticleBt"
                  name="createArticleBt"
                  type="submit"
                >
                  Publish Article
                </button>
              </div>
            </form>
          </div>
        </div>
      </EditArticle>
    </div>
  </EditArticlePage>
</Connect(EditArticlePage)>
`;
